addons/a11y/src/components/A11YPanel.tsx

```diff
@@ -49,228 +48,133 @@ const Centered = styled.span<{}>({
   height: '100%',
 });
 
-interface InitialState {
-  status: 'initial';
-}
-
-interface ManualState {
-  status: 'manual';
-}
-
-interface RunningState {
-  status: 'running';
-}
-
-interface RanState {
-  status: 'ran';
-  passes: Result[];
-  violations: Result[];
-  incomplete: Result[];
-}
-
-interface ReadyState {
-  status: 'ready';
-  passes: Result[];
-  violations: Result[];
-  incomplete: Result[];
-}
-
-interface ErrorState {
-  status: 'error';
-  error: unknown;
-}
-
-type A11YPanelState =
-  | InitialState
-  | ManualState
-  | RunningState
-  | RanState
-  | ReadyState
-  | ErrorState;
-
-interface A11YPanelProps {
-  active: boolean;
-  api: API;
-}
-
-export class A11YPanel extends Component<A11YPanelProps, A11YPanelState> {
-  state: A11YPanelState = {
-    status: 'initial',
-  };
-
-  componentDidMount() {
-    const { api } = this.props;
-
-    api.on(EVENTS.RESULT, this.onResult);
-    api.on(EVENTS.ERROR, this.onError);
-    api.on(EVENTS.MANUAL, this.onManual);
-  }
-
-  componentDidUpdate(prevProps: A11YPanelProps) {
-    // TODO: might be able to remove this
-    const { active } = this.props;
-
-    if (!prevProps.active && active) {
-      // removes all elements from the redux map in store from the previous panel
-      store.dispatch(clearElements());
-    }
-  }
-
-  componentWillUnmount() {
-    const { api } = this.props;
-
-    api.off(EVENTS.RESULT, this.onResult);
-    api.off(EVENTS.ERROR, this.onError);
-    api.off(EVENTS.MANUAL, this.onManual);
-  }
-
-  onResult = ({ passes, violations, incomplete }: AxeResults) => {
-    this.setState(
-      {
-        status: 'ran',
-        passes,
-        violations,
-        incomplete,
-      },
-      () => {
-        setTimeout(() => {
-          const { status } = this.state;
-          if (status === 'ran') {
-            this.setState({
-              status: 'ready',
-            });
-          }
-        }, 900);
+type Status = 'initial' | 'manual' | 'running' | 'error' | 'ran' | 'ready';
+
+export const A11YPanel: React.FC = () => {
+  const [status, setStatus] = React.useState<Status>('initial');
```

Comment:
We can use `useAddonState` perhaps, to have it persist automatically by storing the state in the storybook global state?

Reply:
Yep, good idea